lib/tag-finder.coffee

```diff
@@ -10,23 +11,38 @@ class TagFinder
   constructor: (@editor) ->
     @tagPattern = /(<(\/?))([^\s>]+)([\s>]|$)/
     @wordRegex = /[^>\r\n]*/
-    @tagSelector = SelectorCache.get('meta.tag | punctuation.definition.tag')
-    @commentSelector = SelectorCache.get('comment.*')
 
   patternForTagName: (tagName) ->
     tagName = _.escapeRegExp(tagName)
     new RegExp("(<#{tagName}([\\s>]|$))|(</#{tagName}>)", 'gi')
 
   isRangeCommented: (range) ->
-    scopes = @editor.scopeDescriptorForBufferPosition(range.start).getScopesArray()
-    @commentSelector.matches(scopes)
+    @scopesForPositionMatchRegex(range.start, COMMENT_SELECTOR_REGEX)
 
   isTagRange: (range) ->
-    scopes = @editor.scopeDescriptorForBufferPosition(range.start).getScopesArray()
-    @tagSelector.matches(scopes)
+    @scopesForPositionMatchRegex(range.start, TAG_SELECTOR_REGEX)
 
   isCursorOnTag: ->
-    @tagSelector.matches(@editor.getLastCursor().getScopeDescriptor().getScopesArray())
+    @scopesForPositionMatchRegex(@editor.getCursorBufferPosition(), TAG_SELECTOR_REGEX)
+
+  scopesForPositionMatchRegex: (position, regex) ->
```

Comment:
It seems reasonable to not use scope descriptors here because they perform multiple grammar scope lookups and allocate many more objects due to the use of the old `TokenIterator`. We could change `scopeDescriptorForBufferPosition` in core but that would still be less performant because it would need to perform one lookup for every tag that gets opened.

With this method we perform a minimal amount of grammar scope lookups at the cost of some duplication. This is not ideal, especially since we may get rid of first-mate for some grammars soon. Still, I decided to do it anyways as it feels like we'll end up creating richer APIs for tree-sitter no matter what. Thoughts, @nathansobo? 💭